api/http_lifecycle_test.go

```diff
@@ -11,23 +11,20 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-type NopEndpointSource struct {
-	Root string
-}
-
-func (e *NopEndpointSource) Endpoint(op string) config.Endpoint {
-	return config.Endpoint{Url: e.Root}
-}
-
 var (
-	source = &NopEndpointSource{"https://example.com"}
+	testConfig = NewTestConfig()
```

Comment:
Hm... I'd rather not have this be a value shared between tests, since there are still [some][1] [mutative][2] [functions][3] on the `*config.Configuration` type, which could introduce implicit order-based failures.

[1]: https://github.com/git-lfs/git-lfs/blob/769a726b82bf5e1b6c8f43f35d3ec004b385bb78/config/config.go#L223-L226
[2]: https://github.com/git-lfs/git-lfs/blob/769a726b82bf5e1b6c8f43f35d3ec004b385bb78/config/config.go#L302-L305
[3]: https://github.com/git-lfs/git-lfs/blob/769a726b82bf5e1b6c8f43f35d3ec004b385bb78/config/config.go#L436-L457